tensorflow/core/platform/env.h

```diff
@@ -632,7 +632,22 @@ namespace register_file_system {
 
 template <typename Factory>
 struct Register {
-  Register(Env* env, const std::string& scheme) {
+  Register(Env* env, const std::string& scheme, bool legacy) {
+    // TODO(yongtang): Remove legacy file system registration for hdfs/s3/gcs
+    // after TF 2.6+.
+    if (legacy) {
+      const char* enable_legacy_env = getenv("TF_ENABLE_LEGACY_FILESYSTEM");
+      string enable_legacy =
+          enable_legacy_env ? absl::AsciiStrToLower(enable_legacy_env) : "";
+      if (!(enable_legacy == "true" || enable_legacy == "1")) {
+        return;
+      }
+      LOG(WARNING) << "Legacy file system for '" << scheme << "' is deprecated"
+                   << " and will be removed in tensorflow 2.6 or higher."
```

Comment:
@yongtang aren't we looking at TensorFlow 2.5 release for the filesystem migration?

Reply:
@kvignesh1420 If the PR can be merged before upcoming 2.5, then hdfs will be deprecated in 2.5 (with the option of using `TF_ENABLE_LEGACY_FILESYSTEM=1` to keep in legacy mode), in 2.6 or higher we can consider completely remove hdfs.

Reply:
Can we have this env var documented somewhere?